app/controllers/api_controller.rb

```diff
@@ -1,24 +1,24 @@
-class ApiController < ApplicationController  
-	# http_basic_authenticate_with email:ENV["API_AUTH_EMAIL"], :only => [:signup, :signin, :get_token]  
+class ApiController < ApplicationController
+	# http_basic_authenticate_with email:ENV["API_AUTH_EMAIL"], :only => [:signup, :signin, :get_token]
 	#make sure any request that is not a signup, signin, or get_token has it's authtoken checked
-	before_filter :check_for_valid_authtoken, :except => [:signup, :signin, :get_token]	
+	before_filter :check_for_valid_authtoken, :except => [:signup, :signin, :get_token, :location]
 
 	def signin
 	  if request.post?
 	    if params && params[:email]
 	      user = User.where(:email => params[:email]).first
```

Comment:
I know you didn't write this but this should be fixed... Shouldn't be calling doing .where (an expensive query) just to grab .first
Using .find_by is probably better in this case